templates/editAll.html

```diff
@@ -0,0 +1,252 @@
+$def with (selectedAY, fullMountingPlan)  
+
+$ title_string = 'Edit All Mountings and Quotas'
+
+$var title:$title_string
+$ home = ['/', 'Home']
+$ editAll = ['#', title_string]
+
+$var hierarchy = [home, editAll]
+
+<script type="text/javascript">
+    var selectedModuleCount = 0;
+
+    $$(document).ready(function(){
+        //Load the list of modules into the 'Select module to edit' input
+        var moduleSource = [];
+        var moduleCodeAndName = "";
+        $for subplan in fullMountingPlan:
+            moduleCodeAndName = "$subplan[0]"+" - "+"$subplan[1]"
+            moduleSource.push(moduleCodeAndName);
+        $$("input.typeahead").typeahead({
+            minLength: 1,
+            source: moduleSource,
+            updater: function(selectedModule) {
+                var selectedModuleCode = selectedModule.split(" ")[0];
+                var isModuleEdited = selectedModuleCode + "_isEdited";
+                showModuleInTable(selectedModuleCode, isModuleEdited);
+            }
+        })
+    });
+
+    //Show a module in the edit table
+    function showModuleInTable(selectedModuleCode, isModuleEdited) {
+        selectedModuleCount++;
+        var moduleRow = selectedModuleCode + "_row";
+        document.getElementById(moduleRow).style.display = "table-row";
+        document.getElementById(isModuleEdited).value = "True";
+        if (selectedModuleCount > 0) {
+            document.getElementById("edit-all-table-no-modules").style.display = "none";
+        }
+    }
+
+    //Remove a module from the edit table
+    function removeModuleFromTable(moduleRow, isModuleEdited) {
+        selectedModuleCount--;
+        document.getElementById(moduleRow).style.display = "none";
+        document.getElementById(isModuleEdited).value = "False";
+        if (selectedModuleCount <= 0) {
+            document.getElementById("edit-all-table-no-modules").style.display = "table-row";
+        }
+    }
+
+    //Disable quota input when mounting is unchecked, enable quota input when checked
+    function toggleQuotaState(id) {
+        var state = document.getElementById(id).disabled;
+        document.getElementById(id).disabled = !state;
+    }
+
+    //Add red highlight to mounting and quota if mounting is changed
```

Comment:
Steps taken:
1. Enter `/editall`
2. Select a module to edit mounting/quota for (I used CS1020)
3. Change Sem 1's quota to -1.
4. Press the Save button
5. Alert prompts 'Are you sure you want to save your changes?'. That invalid input's cell is shaded red.
6. Press OK.

Results/Comment:
1. Using red color for highlighting changes **was very misleading**. I thought there was something wrong with my inputs or something whenever I applied a change to an input (especially when I checked/unchecked the mounting checkbox). Red color *does* bring attention to something, but more often than not it signifies danger or something going wrong. Consider using another better color instead.

2. Modifying quota for a module did not cause the color change to be applied in real time. The color change was only triggered only after I pressed the Save button, which did not match the statement you made that affected cells are highlight in red in real time.

3. When I entered the invalid quota and I pressed 'OK' after the confirmation dialog, the page does nothing without explicit feedback (on why the page is not performing any action). To us it *is* obvious that the page should do nothing after the attempt to submit data with invalid fields, but this SHOULD be an explicit visual feedback to the user nevertheless.

4. Users should probably not be redirected to a different page after a successful attempt to do mass mounting/quota editing. I think it is better to let the users choose to navigate elsewhere themselves if they personally think that they are done with the edit-all-mountings-and-quotas page.

5. Also, please **do not** add JS scripts directly into the HTML page. The reason I request this is because it makes maintaining all the JavaScripts easier with (a) central JS file(s). Should you require targeting specific elements, you can attach classes/IDs to them and use [DOM methods](https://www.w3schools.com/jsref/dom_obj_all.asp). However, if you have strong justification for doing so, then it is totally fine.